internal/defence/limiter.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package defence
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"storj.io/storj/internal/sync2"
+)
+
+// Attacker stores information about attacker entity
+type Attacker struct {
+	limiter *rate.Limiter
```

Comment:
Why is this a pointer?

Reply:
`rate.NewLimiter` returns a pointer and all the methods are defined for a pointer, so I guess that's for that despite a value would work too.